ledger/src/ui/test/signer_authorization/test_signer_authorization.c

```diff
@@ -0,0 +1,449 @@
+}
+
+int cx_ecfp_init_public_key(cx_curve_t curve,
+                            unsigned char *rawkey,
+                            unsigned int key_len,
+                            cx_ecfp_public_key_t *key) {
+    assert(NULL != rawkey);
+    assert(NULL != key);
+    assert(CX_CURVE_256K1 == curve);
+    assert(65 == key_len);
+
+    memcpy(key->W, rawkey, key_len);
+    key->W_len = key_len;
+
+    return 0;
+}
+
+int cx_ecdsa_verify(cx_ecfp_public_key_t *key,
+                    int mode,
+                    cx_md_t hashID,
+                    unsigned char *hash,
+                    unsigned int hash_len,
+                    unsigned char *sig,
+                    unsigned int sig_len) {
+    // We don't need to perform an actual verification for this test, we just
+    // assert that this was called with the correct parameters and verify wheter
+    // or not one of the authorized keys and the corresponding mock signature
+    // were provided
+    assert(key == &G_sigaut_ctx.pubkey);
+    assert(hash == G_sigaut_ctx.auth_hash);
+    if (is_key_authorized(key)) {
+        return (0 ==
+                memcmp(sig, get_signature(key), TEST_SIGAUT_SIGNATURE_LEN));
+    }
+    return 0;
+}
+
+void assert_sigaut_ctx_reset(sigaut_t *sigaut_ctx) {
+    assert(sigaut_state_wait_signer_version == sigaut_ctx->state);
+    ASSERT_STRUCT_CLEARED(sigaut_signer_t, sigaut_ctx->signer);
+    ASSERT_ARRAY_CLEARED(sigaut_ctx->authorized_signer_verified);
+    ASSERT_STRUCT_CLEARED(cx_sha3_t, sigaut_ctx->auth_hash_ctx);
+    ASSERT_STRUCT_CLEARED(cx_ecfp_public_key_t, sigaut_ctx->pubkey);
+    ASSERT_ARRAY_CLEARED(sigaut_ctx->buf);
+    ASSERT_ARRAY_CLEARED(sigaut_ctx->auth_hash);
+}
+
+void test_reset_signer_authorization() {
+    printf("Test reset signer authorization...\n");
+
+    reset_signer_authorization(&G_sigaut_ctx);
+    assert_sigaut_ctx_reset(&G_sigaut_ctx);
+}
+
+void test_init_signer_authorization() {
+    printf("Test init signer authorization...\n");
+
+    init_signer_authorization();
+    sigaut_signer_t *signer = get_authorized_signer_info();
+    assert(NULL != signer);
+    ASSERT_MEMCMP(PARAM_INITIAL_SIGNER_HASH, signer->hash, HASHSIZE);
+    assert(PARAM_INITIAL_SIGNER_ITERATION == signer->iteration);
+}
+
+void test_op_sigaut_get_current() {
+    printf("Test OP_SIGAUT_GET_CURRENT...\n");
+
+    unsigned int rx;
+    reset_signer_authorization(&G_sigaut_ctx);
+    // OP_SIGAUT_GET_CURRENT
+    SET_APDU("\x80\x81\x00", rx);
+    assert((3 + sizeof(sigaut_signer_t)) ==
+           do_authorize_signer(rx, &G_sigaut_ctx));
+
+    ASSERT_APDU("\x80\x81\x00" PARAM_INITIAL_SIGNER_HASH "\x00\x01");
+}
+
+void test_op_sigaut_sigver() {
+    printf("Test OP_SIGAUT_SIGVER...\n");
+
+    unsigned int rx;
+    reset_signer_authorization(&G_sigaut_ctx);
+    G_sigaut_ctx.state = sigaut_state_wait_signer_version;
+
+    // OP_SIGAUT_SIGVER + hash + iteration
+    SET_APDU("\x80\x81\x01" PARAM_INITIAL_SIGNER_HASH "\x00\x02", rx);
```

Comment:
You probably want to try and use a different signer hash to test this since it'll usually be the case, right?